src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBulkRecordReader.java

```diff
@@ -190,6 +211,18 @@ void setupParser() throws IOException, AsyncApiException {
     resultIdIndex++;
   }
 
+  public InputStream getQueryResultStream(BulkConnection bulkConnection) throws SalesforceQueryExecutionException {
+    try {
+      return bulkConnection.getQueryResultStream(jobId, batchId, resultIds[resultIdIndex]);
+    } catch (AsyncApiException exception) {
+      LOG.error("The bulk query job {} failed.", jobId);
```

Comment:
If you catch it and log it and rethrow, it is considered as an anti-pattern, ref: https://stackoverflow.com/questions/6639963/why-is-log-and-throw-considered-an-anti-pattern

Reply:
Same comment applies to all places in PR.

Reply:
Logged as a warning